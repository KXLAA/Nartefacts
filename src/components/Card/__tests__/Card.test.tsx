import { render } from '@testing-library/react'

import { Card } from '..'

const albumArt =
  'https://ucarecdn.com/c515e4b5-a5bb-44a3-85d0-f139497e0de0/forbrokenearsep_tems.jpeg'

describe('<Card />', () => {
  it('should render card with image', () => {
    const { getByAltText } = render(<Card albumArt={albumArt} />)
    const image = getByAltText('album art')
    expect(image).toBeInTheDocument()
    expect(image).toBeTruthy()
  })

  it('should render card  correctly', () => {
    const { container } = render(<Card albumArt={albumArt} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
