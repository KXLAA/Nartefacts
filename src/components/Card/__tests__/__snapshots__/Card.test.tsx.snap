// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render card  correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 32px;
}

.c1 {
  background: linear-gradient( to bottom,rgba(0,0,0,0.5),rgba(0,0,0,0.5) );
  width: 100%;
  height: 122px;
  border: 4px solid #363636;
  border-radius: 4px;
}

.c2 {
  border: 4px solid #363636;
  border-radius: 4px;
}

.c2 span {
  border-radius: 4px;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
      />
      <img
        alt="album art"
        data-nimg="responsive"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
</div>
`;
